app/services/fastly_vcl/whitelisted_params.rb

```diff
@@ -0,0 +1,58 @@
+module FastlyVCL
+  # Handles updates to our VCL snippet on Fastly that whitelists params
+  class WhitelistedParams
+    VCL_REGEX_START = "^(".freeze
+    VCL_REGEX_END = ")$".freeze
+    FILE_PARAMS = YAML.load_file("config/fastly/whitelisted_params.yml").sort.freeze
+
+    class << self
+      def update
+        fastly = Fastly.new(api_key: ApplicationConfig["FASTLY_API_KEY"])
+        service = fastly.get_service(ApplicationConfig["FASTLY_SERVICE_ID"])
+        latest_version = service.version
+
+        snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                     latest_version.number,
+                                     "Whitelist certain querystring parameters")
+
+        unless params_outdated?(snippet.content)
+          Rails.logger.info("No Fastly VCL updates needed for version #{latest_version.number}.")
+          return
+        end
+
+        new_version = latest_version.clone
+        new_snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                         new_version.number,
+                                         "Whitelist certain querystring parameters")
```

Comment:
It's probably obvious, but I want to make it explicitly clear that the few people with Fastly production access can **NOT** ever change the name of the VCL script in the GUI or this will break.

AFAIK, it's not a big deal, because we haven't changed it since day 1 and very few people even have the access to do that. The gem requires the snippet `name`.

We can skip the gem method and get the snippet a few other ways, but I'd like to stick to the gem for now and I can quickly fix it if/when it becomes an issue.